File: app/src/main/java/kashyap/genovatest/Splash.java
package kashyap.genovatest;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;

import kashyap.genovatest.connect.ConnectivityReceiver;

public class Splash extends AppCompatActivity implements ConnectivityReceiver.ConnectivityReceiverListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        checkConnection();
        init();
    }

    private void init() {
        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(Splash.this, HomePage.class));
                finish();
            }
        }, 3000);
    }

    @Override
    public void onNetworkConnectionChanged(boolean isConnected) {
        showSnack(isConnected);
    }
    private void checkConnection() {
        boolean isConnected = ConnectivityReceiver.isConnected();
        showSnack(isConnected);
    }

    @Override
    protected void onResume() {
        super.onResume();
        IntentFilter intentFilter = new IntentFilter();
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(new ConnectivityReceiver(), intentFilter);
        MyApplication.getInstance().setConnectivityListener(this);

    }


    private void showSnack(boolean isConnected) {
        if (isConnected){
            findViewById(R.id.animation_vie).setVisibility(View.GONE);
            System.out.println("Gone");
        }else{
            findViewById(R.id.animation_vie).setVisibility(View.VISIBLE);
            System.out.println("NoGone");
        }
    }
}

File: app/src/main/java/kashyap/genovatest/Splash.kt
package kashyap.genovatest

import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import android.os.Handler
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import kashyap.genovatest.connect.ConnectivityReceiver
import kashyap.genovatest.connect.ConnectivityReceiver.ConnectivityReceiverListener

class Splash : AppCompatActivity(), ConnectivityReceiverListener {
    var mContext: Context? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        mContext = this
        Static_Catelog.setStringProperty(mContext as Splash, "added", "no")
        checkConnection()
        init()
    }

    private fun init() {
        val handler = Handler()
        handler.postDelayed({
            Intent(this@Splash, HomePage::class.java).putExtra("go", "no")
            startActivity(Intent(this@Splash, HomePage::class.java))
            finish()
        }, 3000)
    }

    override fun onNetworkConnectionChanged(isConnected: Boolean) {
        showSnack(isConnected)
    }

    private fun checkConnection() {
        val isConnected: Boolean = ConnectivityReceiver.isConnected
        showSnack(isConnected)
    }

    override fun onResume() {
        super.onResume()
        val intentFilter = IntentFilter()
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(ConnectivityReceiver(), intentFilter)
        MyApplication.Companion.instance?.setConnectivityListener(this)
    }

    private fun showSnack(isConnected: Boolean) {
        if (isConnected) {
            findViewById<View>(R.id.animation_vie).visibility = View.GONE
            println("Gone")
        } else {
            findViewById<View>(R.id.animation_vie).visibility = View.VISIBLE
            println("NoGone")
        }
    }
}